src/main/scala/nl.knaw.dans.easy.file2bag/FilesXml.scala

```diff
@@ -31,27 +31,41 @@ object FilesXml extends DebugEnhancedLogging {
     // easy-deposit-api does not supply rights at all
     // so here we don't supply a default
     // https://github.com/DANS-KNAW/easy-deposit-api/blob/eef71618e8b776fb274da123f8011510499c741a/src/main/scala/nl.knaw.dans.easy.deposit/docs/FilesXml.scala#L40-L42
-    val itemContent = if (accessRights.isBlank) Seq[Node]()
-                      else <accessibleToRights>{ accessRights }</accessibleToRights>
-    val newItem =
-        <file filepath={ "data/" + destinationPath }>
-          <dcterms:format>{ mimeType }</dcterms:format>
-          { itemContent }
-        </file>
-    logger.info(newItem.toOneLiner)
+    var formatTagPrefix = "dcterms"
+    val oldFilesXmlWithPossiblyAddedNamespace =
+      if ((oldFilesXml \ "file" \ "format").nonEmpty) {
+        formatTagPrefix = (oldFilesXml \ "file" \ "format").head.prefix
+        oldFilesXml
+      }
+      else {
+        if (oldFilesXml.scope.getURI(formatTagPrefix) == null) {
+          oldFilesXml.copy(scope = NamespaceBinding(formatTagPrefix, "http://purl.org/dc/terms/", oldFilesXml.scope))
+        }
+        else
+          oldFilesXml
+      }
+    val accessibleTo = if (accessRights.isBlank) ""
+                       else s"<accessibleToRights>$accessRights</accessibleToRights>"
+    val newFileElement =
+      XML.loadString(
+        s"""<file filepath="data/$destinationPath">
+           <$formatTagPrefix:format>$mimeType</$formatTagPrefix:format>
+           $accessibleTo
+        </file>""")
+    logger.info(newFileElement.toOneLiner)
```

Comment:
Please don't use formatted string to construct XML dynamically. To construct XML with dynamic content you can look at this [example](https://github.com/DANS-KNAW/easy-fedora2vault/blob/c24e92bf7834472f8d9bb8a5efd70a6d967aeb73/src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala#L282-L287).

Reply:
Why not? XML.loadString expects a String, and it checks that the result is valid XML.

Reply:
Why postpone parsing to runtime while you can have it at compile time? When using the XML syntax, the IDE warns you while typing. These warnings are usually easier to understand. Also: the sooner errors are detected, the lower the effort to fix. You also get the benefit of syntax highlighting, what makes the code easier to read and also may raise visual alarms while typing.

Reply:
It is not only element value (`accessRights`) that is parameterized here but also the `namespace prefix` of `format` element (`dcterms/dct`).  I don't think that you can parameterize the prefix in Elem object.